lib/distribute.c

```diff
@@ -250,23 +258,25 @@ DEFUN (distribute_list,
        "Interface name\n")
 {
 	int prefix = (argv[1]->type == WORD_TKN) ? 1 : 0;
-
 	/* Check of distribute list type. */
 	enum distribute_type type = argv[2 + prefix]->arg[0] == 'i'
 					    ? DISTRIBUTE_V4_IN
 					    : DISTRIBUTE_V4_OUT;
 
 	/* Set appropriate function call */
-	void (*distfn)(const char *, enum distribute_type, const char *) =
+	void (*distfn)(const char *, enum distribute_type,
+		       const char *, struct distribute_ctx *) =
 		prefix ? &distribute_list_prefix_set : &distribute_list_set;
+	struct distribute_ctx *ctx =
+		(struct distribute_ctx *)listnode_head(dist_ctx_list);
```

Comment:
This hack should be fine until we convert distribute-lists to the new northbound model, in which case we'll have a much better infrastructure to handle this problem.

Reply:
I am curious to know how you will do that.